### ISSUE TITLE
gimsatul reigns!

### ISSUE THREAD

**Issue by a1880** [arminbiere/gimsatul#2]
Thank you for this masterpiece!

I ported gimsatul to _Microsoft Visual C 2019_ and tested it with a tough example from matrix algebra.
The example has a lot of XOR gates with 15 inputs each. 
Unlike all other 30+ SAT solvers in my reach, gimsatul was able to find a solution.

My questions:
What is the recommended number of threads to use?
(I have a four-CPU machine with eight cores, but `--threads=4` was faster than `--threads=8`)
Is it possible to enforce deterministic behaviour?
(runtimes varied between 185s and an hour for the same example)
Has `--phase=0` the same semantics as in your other solvers?
(most of my input variables have a 0 value in the final solution)

Greetings!

Axel Kemper

**Comment by arminbiere** [comment]
We are still in the process of developing an understanding of how the various features fit together. After a six week tour de force to get this solver off the ground and ready for the competition we used another week for some preliminary experiments which will hopefully be featured in an upcoming paper. It seems that our portfolio strategy is not really important and the solver gives good scalability even without any extra diversification (which is a bit unexpected).  The proof generation part works nicely too. 

In these experiments (we only did SAT Competition 2021 main track as testing a multi-core solver needs quite some resources) we got best results with 32 virtual cores, which was slightly faster than the 16 real cores we had.  Our portfolio repeats after 12 threads.  Unfortunately this solver his very very very non-deterministic and I also do not see an easy way to change that, but clearly is a feature we would want to have.
